### ISSUE TITLE
RAM and keyboard device definition

### ISSUE THREAD

**Issue by DocInTheSky** [expired6978/CUEORGBPlugin#7]
Hello! First of all let me thank you for your work which is amazing and very promising for my purposes.

So, I was looking for a solution to my problem where my GSkill RAM doesn't work with iCUE and every other device of my build does. I was happy to find your plugin which should be my solution but I haven't been able to make my ram work.

Both of my sticks show up and work on OpenRGB:
![image](https://user-images.githubusercontent.com/50404344/128609914-9e891ffa-e360-46c5-a3ef-80c1ae928975.png)

But when they go to iCUE they showed up as keyboards, and with no LEDs detected in there.
I tried creating a "DRAM" default on the settings.json file which at least made it load the images that I selected so that was a little step, but still no LEDs on the devices in iCUE even tho I did this (I got the LED names as they are on OpenRGB):

![image](https://user-images.githubusercontent.com/50404344/128610006-2f9bc631-7ce2-4ada-bdf1-3ef48c7d5e09.png)

I tried creating the device also on devices.json with no luck, trying with either the 8 LEDs or just 1 to try:
![image](https://user-images.githubusercontent.com/50404344/128610036-7e151cbb-ed1b-479f-b5ee-ae4adfaf4ee3.png)

My keyboard did work with 2 keys as I saw that they are the only ones defined but I'm not sure on how to proceed to fully define the whole keyboard but it's something I'm willing to do both with my RAM and keyboard. Any help on this?

If you're open for a chat on Discord please let me know! :)

**Comment by DocInTheSky** [comment]
@expired6978 Bump

**Comment by expired6978** [comment]
The devices are set as keyboards by default because they seem like the only device that uses the polygons for displaying the LED. Other device types are just lists of things. 

I think the issue is you've just named your zones incorrectly. You can see in OpenRGB your device only has one zone and is named "Unknown", the name of the LEDs does not matter, it's just an array of them. It seems like you cannot "resize" your zone either so OpenRGB must already know how many LEDs your device has, so naming your zone "Unknown" should show the LEDs.

Edit: You should only need one View and one Zone. You have two devices with the exact same name so they should use the same configuration

**Comment by DocInTheSky** [comment]
Yes, right now I have gotten it half-working with one zone and one view but it works pretty funky, half the time when using scenes it doesn't respond and when trying to control individual leds it just doesn't work. Also I have seen that some times it thinks that leds from both sticks are the same led and they get controlled at the same time even tho im trying to set them individually.

If you have time, could we discuss it on discord?

**Comment by DocInTheSky** [comment]
This is my config right now, but as I said it doesn't fully work

![image](https://user-images.githubusercontent.com/50404344/129076472-04331c0a-4e08-4c24-9bba-e4cd16a42b28.png)

**Comment by DocInTheSky** [comment]
I have sent you an email, please check it if possible :)

**Comment by H3rian** [comment]
Hey @DocInunez have you solved? i have the same issue with Trident ZRGB

**Comment by Dravenoid** [comment]
Hi there! I was wondering if this has been figured out? 
I was planning to figure it out myself and make the same thing all this from scratch until I stumbled upon this.

**Comment by jojorne** [comment]
```json
	},
	{
		"Name": "ENE DRAM",
		"InheritDefault": false,
		"Zones": [
			{"Zone": "DRAM"},
			{"Zone": "Unknown"}
		],
		"Views": []
	}
]
```` 
I added this to the `devices.json` file and changed the mode to direct.
Note that if you have scenes on, you cannot change colors inside iCue.

**Comment by jojorne** [comment]
```json
	},
	{
		"Name": "ENE DRAM",
		"InheritDefault": false,
		"Zones": [
			{"Zone": "DRAM"},
			{"Zone": "Unknown"}
		]
	}
]
```
I added this to the `devices.json` file and changed the `Mode` to `Direct`.
Note that if you have scenes on, you cannot change the LEDs in iCue.

**Comment by QuiteAnthony** [comment]
Anyone figured out how to get Gskill ram rgb working ? Will there be some updates for CUEORGBPlugin ?
